src/sas/qtgui/Utilities/TabbedModelEditor.py

```diff
@@ -743,9 +1038,146 @@ def VR(%(args)s):
+//       file linking to your C model.
+// NOTE: It also has many common constants following the C99 standard, such as
+//       M_PI, M_SQRT1_2, and M_E. Check documentation for full list.
+
+'''
+
+C_PD_TEMPLATE = '''\
+static double
+form_volume({poly_args}) // Remove arguments as needed
+{{
+    return 0.0*{poly_arg1};
+}}
+'''
+
+C_TEMPLATE = """\
+static double
+radius_effective(int mode) // Add arguments as needed
+{{
+    switch (mode) {{
+    default:
+    case 1:
+    // Define effective radius calculations here...
+    return 0.0;
+    }}
+}}
+
+static void
+Fq(double q, 
+   double *F1,
+   double *F2,
+   {args}) // Remove arguments as needed
+{{
+    // Define F(Q) calculations here...
+    //IMPORTANT: You do not have to define Iq if your model uses Fq for
+    //    beta approximation; the *F2 value is F(Q)^2 and equivalent to
+    //    the output of Iq. You may use Fq instead of Iq even if you do
+    //    not need F(Q) (*F1) for beta approximation, but this is not recommended.
+    //    Additionally, you must still define Iqac or Iqabc if your
+    //    model has orientation parameters.
+    *F1 = 0.0;
+    *F2 = 0.0;
+}}
+
+static double
+Iq(double q,
+   {args}) // Remove arguments as needed
+{{
+    // Define I(Q) calculations here for models independent of shape orientation
+    // IMPORTANT: Only define ONE calculation for I(Q): either Iq, Iqac, or Iqabc;
+    //    remove others.
+    // TO USE: Convert your copied Python code to C below and uncomment it
+    // Ensure that you delete the I(Q) function in the corresponding Python file.
+
+    {Iq}
+    return 1.0;
+}}
+
+static double
+Iqac(double qab,
+     double qc,
+     {args}) // Remove arguments as needed
+{{
+    // Define I(Q) calculations here for models dependent on shape orientation in
+    // which the shape is rotationally symmetric about *c* axis.
+    // Note: *psi* angle not needed for shapes symmetric about *c* axis
+    // IMPORTANT: Only define ONE calculation for I(Q): either Iq, Iqac, Iqabc, or Iqxy;
+    //   remove others.
+    return 1.0;
+}}
+
+static double
+Iqabc(double qa,
+      double qb,
+      double qc,
+      {args}) // Remove arguments as needed
+{{
+    // Define I(Q) calculations here for models dependent on shape orientation in
+    // all three axes.
+    // IMPORTANT: Only define ONE calculation for I(Q): either Iq, Iqac, Iqabc, or Iqxy;
+    //   remove others.
+    return 1.0;
+}}
+
+static double
+Iqxy(double qx,
+     double qy,
+     {args}) // Remove arguments as needed
+{{
+    // Define I(Q) calculations here for 2D magnetic models.
+    // WARNING: The use of Iqxy is generally discouraged; Use Iqabc instead
+    //    for its better orientational averaging and documentation for details.
+    // IMPORTANT: Only define ONE calculation for I(Q): either Iq, Iqac, Iqabc, or Iqxy;
+    //    remove others.
+    
+    return 1.0;
+}}
+"""
+
 if __name__ == '__main__':
```

Comment:
Not sure this is really needed. We no longer test those widgets as separate windows, but rather inside SasView. Please consider removing the whole conditional